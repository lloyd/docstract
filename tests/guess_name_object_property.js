// We should be able to guess both the name and type of this property.
// (a function named "destroy")

    /**
     * Stops the server from accepting new connections. This function is
     * asynchronous, the server is finally closed when the server emits a
     * 'close' event.
     */
    destroy: function() {
        if (this._process) {
            this._process.kill();
            this._process = null;
        }
        this.stdin.destroy();
        this.stdout.destroy();
        this.stderr.destroy();
        this._removeAllListeners("exit")
        delete processes[this._guid];
    }
