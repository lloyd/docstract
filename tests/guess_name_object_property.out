{
  "filename": "tests/guess_name_object_property.js", 
  "functions": [
    {
      "desc": "Stops the server from accepting new connections. This function is\nasynchronous, the server is finally closed when the server emits a\n'close' event.", 
      "name": "destroy", 
      "source_lines": [
        4, 
        8
      ]
    }
  ], 
  "module": "guess_name_object_property"
}
